src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -434,6 +437,160 @@ Map<String, Object> buildNLPResult(Map<String, Object> processorMap, List<?> res
         return result;
     }
 
+    /**
+     * This method traverses the ProcessMap generated with IngestDocument to filter out the entries that can be skipped for inference.
+     * @param existingSourceAndMetadataMap SourceAndMetadataMap of existing Document
+     * @param sourceAndMetadataMap SourceAndMetadataMap of IngestDocument
+     * @param processMap processMap built from ingestDocument
+     * @return filtered ProcessMap with only entries that require model inferencing
+     */
+
+    protected Map<String, Object> filterProcessMap(
```

Comment:
Defining a shared method in the parent class may not be ideal for simplifying unit testing. A better approach would be to create a separate class and inject it into the processor. This allows the injected instance to be easily mocked during unit testing, making the process of writing unit tests much more straightforward.